.travis.yml

```diff
@@ -1,27 +1,127 @@
+######## Testing minimal compiler requirements ########
+# GCC          6.1.0        here: 6.4.0
+# Clang        4.0          here: 3.8,3.9 not compiling
+# Apple Clang  9            here: clang-900.0.39.2
+#######################################################
+
 language: cpp
 sudo: false
-os: linux
-dist: trusty
-compiler: gcc
 
-addons:
-    apt:
-        sources:
+matrix:
+  include:
+########################## OS X #########################
+## test gcc6 - single node/rank with threading backend ##
+    - os: osx
+      osx_image: xcode9.2
+      env:
+        - MATRIX_EVAL="brew install gcc@6 && brew link --force --overwrite gcc@6 && CC=gcc-6 && CXX=g++-6"
+        - BUILD_NAME=cthread WITH_DISTRIBUTED=serial
+      compiler: gcc-6
+
+## test gcc6 - mpi with threading backend ##
+    - os: osx
+      osx_image: xcode9.2
+      env:
+        - MATRIX_EVAL="brew install gcc@6 && brew link --force --overwrite gcc@6 && CC=gcc-6 && CXX=g++-6"
+        - BUILD_NAME=mpi WITH_DISTRIBUTED=mpi
+      compiler: gcc-6
+
+## test clang9 - single node/rank with threading backend ##
+    - os: osx
+      # xcode9/9.1 broken, 9.2 still uses Apple LLVM version 9.0.0 (clang-900.0.39.2)
+      osx_image: xcode9.2
+      env:
+        - MATRIX_EVAL="CC=clang && CXX=clang++"
+        - BUILD_NAME=cthread WITH_DISTRIBUTED=serial
+      compiler: clang
+
+
+## test clang9 - mpi with threading backend ##
+    - os: osx
```

Comment:
Is there a way to give this test a name that clearly describes what it is testing.
If you look at the dashboard for the tests here:
https://travis-ci.org/eth-cscs/arbor/builds/432535277
It would be nice if it was clearly labeled with something like `os x, apple clang, mpi`

Reply:
Will need to research this option

Reply:
I don't know if it is possible, but it would be nice!